### DISCUSSION TITLE
how to log response body via log plugins?

### DISCUSSION THREAD

**Discussion by gracekk** [apache/apisix#5523] [Q&A]
Sometimes I want to save response body details into logs, such as http 400 occurs. Tried to modify http-logger plugins, but seems context of log_by_lua no appropriate. Any suggestions?  Thank you.

**Comment by tokers** [comment]
How did you get the response body? You may have to add the `body_filter` hook to save the response body.

**Comment by syzygyyy** [comment]
Thanks for reply. 
Tried add "body_filter_by_lua_file" partition in tpl.lua to edit ngnix.conf, got errors:

usr/local/openresty/luajit/bin/luajit ./apisix/cli/apisix.lua init_etcd
nginx: [emerg] "body_filter_by_lua_file" directive is duplicate in /usr/local/apisix/conf/nginx.conf:345
/usr/local/openresty/luajit/bin/luajit ./apisix/cli/apisix.lua init


seems  conflict with
body_filter_by_lua_block {
    apisix.http_body_filter_phase()
}

**Reply by tokers** [reply]
You don't add these directly, just implement the method (`body_filter`) in the plugin implementation.
